src/areconnectedthroughproperties.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module utils/arestructuresconnected
+ */
+
+/* globals EventTarget, Event */
+
+/**
+ * Traverses both structures to find out whether there's some object that is shared between both structures.
+ *
+ * @param {*} str1
+ * @param {*} str2
```

Comment:
Shouldn't it be `{Object}` and maybe we could change names to `obj1` and `obj2`? :)

Reply:
Or maybe `{Object|null}` if that's a case?

Reply:
It's hard to be precise in JS types 😅

In TS there are an `object` and an `Object` ( `{}` ) types, where only objects are assignable to the first while all primitive values except `null` and `undefined` can be assigned to the second (all stuff inheriting from the `Object`).

But in our JSDoc system, we use mainly `*` for such types. We use the `Object` only when we accept objects, not primitives. And bear in mind that arrays should be assignable here too 😄

Reply:
OTOH I can take your point, that from this function might expect objects and arrays only.